samtranslator/model/eventsources/push.py

```diff
@@ -47,9 +48,13 @@ def _construct_permission(self, function, source_arn=None, source_account=None,
         :returns: the permission resource
         :rtype: model.lambda_.LambdaPermission
         """
-        lambda_permission = LambdaPermission(self.logical_id + 'Permission' + suffix,
+        if '-' in suffix or '_' in suffix:
```

Comment:
Same here

Reply:
I just changed the condition in the latest commit.